mobx/lib/src/api/observable_list.dart

```diff
@@ -13,5 +16,480 @@ import 'package:mobx/src/core.dart';
+  Iterable<U> whereType<U>() {
+    _atom.reportObserved();
+    return _list.whereType<U>();
+  }
+
+  //------------------ Write Methods -------------------- //
+  @override
+  set first(T value) {
+    final oldValue = _list.first;
+
+    _list.first = value;
+    _notifyChildUpdate(0, value, oldValue);
+  }
+
+  @override
+  set last(T value) {
+    final oldValue = _list.last;
+
+    _list.last = value;
+    _notifyChildUpdate(_list.length - 1, value, oldValue);
+  }
+
+  @override
+  void operator []=(int index, T value) {
+    final oldValue = _list[index];
+
+    if (oldValue != value) {
+      _list[index] = value;
+      _notifyChildUpdate(index, value, oldValue);
+    }
+  }
+
+  @override
+  set length(int value) {
+    _list.length = value;
+    _notifyListUpdate(0, null, null);
+  }
+
+  @override
+  void add(T value) {
+    _list.add(value);
+    _notifyListUpdate(_list.length, [value], null);
+  }
+
+  @override
+  void addAll(Iterable<T> iterable) {
+    _list.addAll(iterable);
+    _notifyListUpdate(0, iterable.toList(growable: false), null);
+  }
+
+  @override
+  void clear() {
+    _list.clear();
+    _notifyListUpdate(0, null, null);
+  }
+
+  @override
+  void fillRange(int start, int end, [T fillValue]) {
+    _list.fillRange(start, end, fillValue);
+    _notifyListUpdate(start, null, null);
+  }
+
+  @override
+  void insert(int index, T element) {
+    _list.insert(index, element);
+    _notifyListUpdate(index, [element], null);
+  }
+
+  @override
+  void insertAll(int index, Iterable<T> iterable) {
+    _list.insertAll(index, iterable);
+    _notifyListUpdate(index, iterable.toList(growable: false), null);
+  }
+
+  @override
+  bool remove(Object value) {
+    final index = _list.indexWhere((_) => _ == value);
+    final didRemove = _list.remove(value);
+
+    if (didRemove) {
+      _notifyListUpdate(index, null, value == null ? null : [value]);
+    }
+
+    return didRemove;
+  }
+
+  @override
+  T removeAt(int index) {
+    final value = _list.removeAt(index);
+    _notifyListUpdate(index, null, value == null ? null : [value]);
+
+    return value;
+  }
+
+  @override
+  T removeLast() {
+    final value = _list.removeLast();
+
+    // Index is _list.length as it points to the index before the last element is removed
+    _notifyListUpdate(_list.length, null, value == null ? null : [value]);
+
+    return value;
+  }
+
+  @override
+  void removeRange(int start, int end) {
+    _list.removeRange(start, end);
+    _notifyListUpdate(start, null, null);
+  }
+
+  @override
+  void removeWhere(bool Function(T element) test) {
+    _list.removeWhere(test);
+    _notifyListUpdate(0, null, null);
+  }
+
+  @override
+  void replaceRange(int start, int end, Iterable<T> replacement) {
+    _list.replaceRange(start, end, replacement);
+    _notifyListUpdate(start, null, null);
+  }
+
+  @override
+  void retainWhere(bool Function(T element) test) {
+    _list.retainWhere(test);
+    _notifyListUpdate(0, null, null);
```

Comment:
shouldn't these kinds of methods provide the removed items as the last argument?

Reply:
agree...added in the latest commit